google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/BigQueryImplTest.java

```diff
@@ -874,71 +877,95 @@ public void testListTableDataWithOptions() {
     assertArrayEquals(TABLE_DATA.toArray(), Iterables.toArray(page.getValues(), List.class));
   }
 
-  @Test
-  public void testCreateQueryJob() {
-    EasyMock.expect(
-            bigqueryRpcMock.create(
-                JobInfo.of(QUERY_JOB_CONFIGURATION_WITH_PROJECT).toPb(), EMPTY_RPC_OPTIONS))
-        .andReturn(COMPLETE_QUERY_JOB.toPb());
-    EasyMock.replay(bigqueryRpcMock);
-    bigquery = options.getService();
-    Job job = bigquery.create(QUERY_JOB);
-    assertEquals(new Job(bigquery, new JobInfo.BuilderImpl(COMPLETE_QUERY_JOB)), job);
+  // The "minimally initialized" Job that lets Job.fromPb run without throwing.
+  private static com.google.api.services.bigquery.model.Job newJobPb() {
+    return new com.google.api.services.bigquery.model.Job()
+        .setConfiguration(
+            new com.google.api.services.bigquery.model.JobConfiguration()
+                .setQuery(new JobConfigurationQuery().setQuery("FOO")));
   }
 
   @Test
-  public void testCreateLoadJob() {
+  public void testCreateJobSuccess() {
+    String id = "testCreateJobSuccess-id";
+    JobId jobId = JobId.of(id);
+    String query = "SELECT * in FOO";
+
+    Capture<com.google.api.services.bigquery.model.Job> jobCapture = EasyMock.newCapture();
     EasyMock.expect(
-            bigqueryRpcMock.create(
-                JobInfo.of(LOAD_JOB_CONFIGURATION_WITH_PROJECT).toPb(), EMPTY_RPC_OPTIONS))
-        .andReturn(COMPLETE_LOAD_JOB.toPb());
+            bigqueryRpcMock.create(EasyMock.capture(jobCapture), EasyMock.eq(EMPTY_RPC_OPTIONS)))
+        .andReturn(newJobPb());
     EasyMock.replay(bigqueryRpcMock);
+
     bigquery = options.getService();
-    Job job = bigquery.create(LOAD_JOB);
-    assertEquals(new Job(bigquery, new JobInfo.BuilderImpl(COMPLETE_LOAD_JOB)), job);
+    assertThat(bigquery.create(JobInfo.of(jobId, QueryJobConfiguration.of(query)))).isNotNull();
+    assertThat(jobCapture.getValue().getJobReference().getJobId()).isEqualTo(id);
   }
 
   @Test
-  public void testCreateCopyJob() {
+  public void testCreateJobWithSelectedFields() {
+    Capture<Map<BigQueryRpc.Option, Object>> capturedOptions = Capture.newInstance();
     EasyMock.expect(
             bigqueryRpcMock.create(
-                JobInfo.of(COPY_JOB_CONFIGURATION_WITH_PROJECT).toPb(), EMPTY_RPC_OPTIONS))
-        .andReturn(COMPLETE_COPY_JOB.toPb());
+                EasyMock.anyObject(com.google.api.services.bigquery.model.Job.class),
+                EasyMock.capture(capturedOptions)))
+        .andReturn(newJobPb());
     EasyMock.replay(bigqueryRpcMock);
+
     bigquery = options.getService();
-    Job job = bigquery.create(COPY_JOB);
-    assertEquals(new Job(bigquery, new JobInfo.BuilderImpl(COMPLETE_COPY_JOB)), job);
+    Job job = bigquery.create(QUERY_JOB, JOB_OPTION_FIELDS);
```

Comment:
It's not so good to have a test that depends on a value defined elsewhere. Could you set the options that you're looking for below here?